src/mixins/object_stacking.mixin.js

```diff
@@ -76,5 +76,48 @@ fabric.util.object.extend(fabric.Object.prototype, /** @lends fabric.Object.prot
       this.canvas.moveTo(this, index);
     }
     return this;
+  },
+
+  /**
+   *
+   * @param {fabric.Object} other object to compare against
+   * @returns {boolean | undefined} if objects do not share a common ancestor or they are strictly equal it is impossible to determine which is in front of the other; in such cases the function returns `undefined`
```

Comment:
not sharing a common ancestor may only mean theyare on different canvases?
In that case can we print a log warn and return false rather than returning true/false/undefined?

Reply:
> I think that it should remain. The result of checking which object is in front of the other is really undefined in this case. false means it is not.